teps/0091-In-toto-layout-creation-via-tekton-cli.md

```diff
@@ -0,0 +1,708 @@
+
+-->
+
+<!--
+This is the title of your TEP.  Keep it short, simple, and descriptive.  A good
+title can help communicate what the TEP is and should be considered as part of
+any review.
+-->
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a TEP and for
+highlighting any additional information provided beyond the standard TEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This proposal adds to improve how tekton chains and in-toto can be used together to perform verification.
+Currently Tekton Chains already produces the in-toto attestations for each taskruns. The next step to use
+this attestation within in-toto is to produce a layout file that can be fed into `in-toto verify` to validate
+the steps that each taskrun took. This proposal seeks to add a new command to Tekton CLI (via library) or third party tool that can be called on 
+an existing task, taskrun, pipeline or pipelinerun and automatically create a specific layout file for ingestion by `in-toto verify`.
+
+
+## Motivation
+
+Currently there is a disconnect between the two tools. Having an automated way to create the layout file breaks the
+barrier to entry. This new addition would allow users to use both Tekton Chains along with in-toto to verify their work.
+Creating a layout file that can be used by in-toto is a non-trivial task. 
+
+Example of a layout file that would be used by in-toto verify: https://github.com/in-toto/in-toto-golang/blob/master/certs/layout.tmpl
+
+
+### Goals
+
+Add a new CLI command or third party tool that would create a in-toto layout file based on a specifying a task, taskrun, pipeline or pipelinerun from within Tekton.
+Determine the format of the layout file that both Tekton and In-toto communities can agree on and would work with `in-toto verify`.
+
+POC has been created: [tkn-intoto-formatter](https://github.com/tap8stry/tkn-intoto-formatter). See below [References (optional)](#references-optional).
+
```

Comment:
is it possible to explain somewhere (e.g. in motivation or maybe use cases) how someone might integrate this into their use of Tekton Pipelines? for example, do we expect that a user would generate the layout file for a Pipeline, then execute that Pipeline, then verify it later, or would we want to build the verification into Tekton itself?